tests/unit/amo/components/TestPage.js

```diff
@@ -711,4 +714,151 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('Tests for SearchForm', () => {
+    it('sets the form action URL', () => {
+      dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+        lang: 'en-GB',
+        store,
+      });
+      render();
+
+      // Header passes a pathname of `/search/` to SearchForm.
+      expect(screen.getByRole('search')).toHaveAttribute(
+        'action',
+        `/en-GB/${CLIENT_APP_FIREFOX}/search/`,
+      );
+    });
+
+    it('changes the URL on search', () => {
+      dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+        lang: 'en-GB',
+        store,
+      });
+      const history = createHistory();
+      const query = 'panda themes';
+      render({ history });
+
+      const pushSpy = jest.spyOn(history, 'push');
+
+      userEvent.type(screen.getByRole('searchbox'), `{selectall}{del}${query}`);
+      userEvent.click(screen.getByRole('button', { name: 'Search' }));
+
+      expect(pushSpy).toHaveBeenCalledWith({
+        pathname: `/en-GB/${CLIENT_APP_FIREFOX}/search/`,
+        query: { q: query },
+      });
+    });
+
+    it('pushes a new route when a suggestion is selected', () => {
+      const url = '/url/to/extension/detail/page';
+      const fakeResult = createFakeAutocompleteResult({ url });
+      const history = createHistory();
+      render({ history });
+
+      const pushSpy = jest.spyOn(history, 'push');
+
+      userEvent.type(screen.getByRole('searchbox'), 'test');
+      dispatchAutocompleteResults({ results: [fakeResult], store });
+      userEvent.click(
+        screen.getByRole('option', {
+          // This is the accessible name for the suggestion.
+          name: 'suggestion-result suggestion-result Go to the add-on page',
+        }),
+      );
+
+      expect(pushSpy).toHaveBeenCalledWith(url);
+    });
+
+    it('parses the URL of a suggestion to push the pathname', () => {
+      const pathname = '/url/to/extension/detail/page';
+      const fakeResult = createFakeAutocompleteResult({
+        url: `https://example.org${pathname}`,
+      });
+      const history = createHistory();
+      render({ history });
+
+      const pushSpy = jest.spyOn(history, 'push');
+
+      userEvent.type(screen.getByRole('searchbox'), 'test');
+      dispatchAutocompleteResults({ results: [fakeResult], store });
+      userEvent.click(
+        screen.getByRole('option', {
+          name: 'suggestion-result suggestion-result Go to the add-on page',
+        }),
+      );
+
+      expect(pushSpy).toHaveBeenCalledWith(pathname);
```

Comment:
nit: these seems to be identical to the previous test, just with this extra line?

Reply:
Good point. I can address that with an `it.each()` test.